packages/wdio-reporter/src/index.ts

```diff
@@ -1,17 +1,42 @@
 import fs from 'fs'
 import fse from 'fs-extra'
-import EventEmitter from 'events'
+import { EventEmitter } from 'events'
 
 import { getErrorsFromEvent } from './utils'
 
-import SuiteStats from './stats/suite'
-import HookStats from './stats/hook'
-import TestStats from './stats/test'
+import SuiteStats, { Suite } from './stats/suite'
+import HookStats, { Hook } from './stats/hook'
+import TestStats, { Test } from './stats/test'
 
-import RunnerStats from './stats/runner'
+import RunnerStats, { Runner } from './stats/runner'
+
+interface WDIOReporterOptions {
+    outputDir: string
+    logFile: string
+    stdout: boolean
+    writeStream: fs.WriteStream
+}
 
 export default class WDIOReporter extends EventEmitter {
-    constructor(options) {
+
+    outputStream: fs.WriteStream
+    failures = []
```

Comment:
can you define type?

Reply:
Did that, however, should failures and retries be optional. or instantiated with 0?